src/fragmentarium/ui/display/Transliteration.tsx

```diff
@@ -1,117 +1,58 @@
   Line,
   Text,
   DollarAndAtLine,
-  RulingDollarLine
+  RulingDollarLine,
+  TextLine,
+  LineNumber,
+  LineNumberRange
 } from 'fragmentarium/domain/text'
-import DisplayToken, { TokenWrapper } from './DisplayToken'
-import { isEnclosure, isShift, isDocumentOrientedGloss } from './type-guards'
-
 import './Display.sass'
-import { Shift, Token } from 'fragmentarium/domain/token'
+import DisplayLineTokens from './LineAccumulator'
 
-function WordSeparator({
-  modifiers: bemModifiers = []
-}: {
-  modifiers?: readonly string[]
-}): JSX.Element {
-  const element = 'Transliteration__wordSeparator'
-  return (
-    <span
-      className={classNames([
-        element,
-        bemModifiers.map(flag => `${element}--${flag}`)
-      ])}
-    >
-      {' '}
-    </span>
-  )
+function formatLineNumber(lineNumber: LineNumber): string {
+  return `(${lineNumberToString(lineNumber)})`
 }
 
-function isCloseEnclosure(token: Token): boolean {
-  return isEnclosure(token) && ['CENTER', 'RIGHT'].includes(token.side)
+function formatLineNumberRange({ start, end }: LineNumberRange): string {
+  return `(${lineNumberToString(start)}-${lineNumberToString(end)})`
 }
-
-function isOpenEnclosure(token: Token): boolean {
-  return isEnclosure(token) && ['CENTER', 'LEFT'].includes(token.side)
+function lineNumberToString({
+  hasPrime,
+  number,
+  prefixModifier,
+  suffixModifier
+}: LineNumber): string {
+  const prefixMod = prefixModifier ? prefixModifier + '+' : ''
+  const prime = hasPrime ? '′' : ''
+  const suffixMod = suffixModifier ? suffixModifier : ''
+  return `${prefixMod}${number}${prime}${suffixMod}`
 }
 
-class LineAccumulator {
-  result: React.ReactNode[] = []
-  private inGloss = false
-  private language = 'AKKADIAN'
-  private enclosureOpened = false
-
-  applyLanguage(token: Shift): void {
-    this.language = token.language
-  }
-
-  pushToken(token: Token): void {
-    if (this.requireSeparator(token)) {
-      this.pushSeparator(this.result)
-    }
-
-    const glossWrapper: TokenWrapper = ({
-      children
-    }: PropsWithChildren<{}>): JSX.Element => (
-      <sup className="Transliteration__DocumentOrientedGloss">{children}</sup>
-    )
-
-    this.result.push(
-      this.inGloss && !isEnclosure(token) ? (
-        <DisplayToken
-          key={this.result.length}
-          token={token}
-          bemModifiers={[this.language]}
-          Wrapper={glossWrapper}
-        />
-      ) : (
-        <DisplayToken
-          key={this.result.length}
-          token={token}
-          bemModifiers={[this.language]}
-        />
-      )
-    )
-    this.enclosureOpened = isOpenEnclosure(token)
-  }
-
-  openGloss(): void {
-    this.inGloss = true
-  }
-
-  closeGloss(): void {
-    this.inGloss = false
-  }
-
-  private requireSeparator(token: Token): boolean {
-    const noEnclosure = !isCloseEnclosure(token) && !this.enclosureOpened
-    return this.result.length === 0 || noEnclosure
-  }
+const lineNumberTypeToComponent = {
```

Comment:
`formatLineNumber` and `formatLineNumberRange` are not components. The name should be changed accordingly,